xcpmd/src/rules.c

```diff
@@ -586,6 +586,25 @@ void delete_rules(void) {
 }
 
 
+//Reruns all hooks for this rule.
+void refresh_rule(struct rule * rule) {
+
+    struct condition *cond;
+
+    if (rule == NULL) {
+        xcpmd_log(LOG_DEBUG, "Couldn't refresh null rule\n");
+        return;
+    }
+
+    //Right now, the only hook is condition on_instantiate().
+    list_for_each_entry(cond, &(rule->conditions.list), list) {
+        if (cond->type->on_instantiate != NULL) {
+            cond->type->on_instantiate(cond);
```

Comment:
Are all the on_instantiate methods written to handle being call multiple times?  I found a case where dar_idle_instantiate() makes libevent/xcpmd go into an infinite loop when called multiple times.

Reply:
I'll PR the fix for `dar_idle_instantiate()` "soon".